### DISCUSSION TITLE
Open Server Panel 6.0.0

### DISCUSSION THREAD

**Discussion by OSPanel** [OSPanel/OpenServerPanel#8] [Анонсы]
### Что нового

- Планировщик заданий
- Конвертер доменных имён
- Автоматическая проверка обновлений
- Модули NVM 1.1.12, RabbitMQ 3.13.1, Nginx-1.26.0 (OpenSSL 3.3.0) и Nginx-1.22.1 (OpenSSL 1.1.1w)
- Дополнение Erlang/OTP (необходимо для RabbitMQ)
- Компилятор Dart Sass 1.75.0
- Вспомогательные утилиты командной строки: bat, brotli, dust, fd, gzip, sass (dart), sd, xh
- Возможность выбора версии Nginx и Node.js из трей-меню
- Возможность запуска проекта из трей-меню (например Node.js приложения)
- Возможность использования модулем среды любого дополнения
- FastCGI версии модулей PHP
- Новый дизайн страниц доменов-примеров доступных после установки программы
- Автоматическое определение кодировки файла HOSTS
- Новый формат работы с командой `osp log ...` (через регулярные выражения)
- Возможность редактирования используемых по умолчанию настроек доменов 
- Возможность задать индивидуальные переменные окружения для каждого проекта 
- Автоматическое открытие электронных писем (по умолчанию выключено)
- Расширение php_zstd для PHP 8.2 и 8.3
- Автоматическая подсветка при выводе журналов в консоли
- Проверка привилегий и статуса UAC для PostgreSQL
- Секция `[services]` в конфиге модулей для возможности проверки состояния служб Windows
- Команда `addons` для листинга имеющихся дополнений
- Команда `node` для управления версиями Node.js
- Команда `status` для вывода общего статуса всех систем
- Опции `php_engine`, `nginx_engine` и `node_engine` для указания движков проекта
- Опции `use_win_terminal` и `need_use_win_terminal` для пользователей Windows Terminal
- Опции `environment` (проекты) и `used_addons_environment` (модули) для проброса сред
- Опции `projects_search_depth` и `projects_search_path` (см. [документацию](https://github.com/OSPanel/OpenServerPanel/wiki/%D0%94%D0%BE%D0%BA%D1%83%D0%BC%D0%B5%D0%BD%D1%82%D0%B0%D1%86%D0%B8%D1%8F))
- Опция `project_url` для возможности задать URL проекта
- Опции меню: `need_start_command`, `show_icons` и `show_hr_after_*`
- Иконки в меню программы (доступны наборы Font Awesome Free Solid & Font Awesome Free Regular)
- Проверка на запуск из PowerShell с выводом сообщения об ошибке
- Модули, дополнения и проекты теперь могут дополнять языковые константы и пункты меню
- Улучшена информативность листинга модулей и доменов в консоли

### Изменения

- Убран статус бета-версии (веб-интерфейс будет добавлен позже)
- Добавление проектов в программу теперь осуществляется по новому принципу (см. документацию)
- Упразднено понятие пассивных модулей (переехали в дополнения)
- Упразднён модуль ControlPanel и файл конфигурации domains.ini
- Perl перемещён в дополнения и более не является модулем
- Perl и доп. наборы данных выведены в отдельные пункты меню в установщике
- Модуль mod_perl теперь по умолчанию отключен в конфиге Apache
- Удалён модуль PHP 7.1 (нестабильная работа)
- Удалены некоторые PHP-расширения: ioncube, sandbox, svn (нестабильная работа)
- Теперь нельзя добавлять в среду консоли модули одного типа (PHP+PHP, MySQL+MySQL и т.д.)
- Переменная окружения `HOME` по умолчанию теперь общая для всех модулей
- По умолчанию при просмотре логов в консоли теперь выводится всё содержимое журнала
- Изменена логика и формат команды просмотра логов в консоли
- Упразднены опции (программа и меню): `hosts_file_encoding` и `need_state`
- Упразднены опции (домены): `auto_configure`, `cgi_dir`, `engine`, `log_format`, `project_use_sys_env` и `ssl_auto_cert`
- Опция `project_add_modules` переименована в `environment`
- Опция `project_add_commands` переименована в `start_command` с заменой предназначения
- Опция `group_projects_by_tld` переименована в `show_projects_in_groups`
- Опция `project_home_dir` переименована в `project_dir`
- Опция `allowed_env_vars` переименована в `allowed_system_env_vars`
- Изменён формат и логика конфигурации трей-меню программы
- Доработан конфиг модулей Apache
- Протокол HTTP/2 включён по умолчанию в Apache и Nginx (кроме Nginx-1.22.1)
- Apache-модуль `proxy_module` активирован по умолчанию
- При работе в консоли больше не требуется `osp reset init` перед `osp use ...` и `osp project ...`
- Модули, дополнения и вспомогательное ПО обновлены до актуальных версий
- Большие изменения и дополнения в документации к программе

### Исправления

- Рабочий процесс PHP мог завершаться с ошибкой после 10000 запросов к веб-серверу
- Программа могла зависнуть при перезапуске какого-либо модуля
- При выключении модуля не удались записи его доменов из HOSTS файла
- При перезапуске модуля не удалялись и накапливались записи доменов с жестко назначенным IP
- Программа могла зависнуть при закрытии с активной HTTP-сессией к API
- Программа могла зависнуть при наличии модулей не совместимых с текущей версией Windows
- Программа не перечитывала настройки среды модуля при его перезапуске
- Слишком маленькая высота пунктов меню в Windows 11  
- Процедура формирования меню программы выполнялась некорректно
- Значение опции `project_add_commands` обрабатывалось неверно и команда могла не выполняться 
- Модули MySQL и MariaDB не запускались при наличии сторонних файлов my.cnf на системном диске 
- Журналы могли выводиться в нечитаемой кодировке при размере файла журнала более 32 КБ

### Важно!

- Не производите установку поверх старой версии!
- Используйте команды `osp sysprep` и `osp cacert init` если вы пропустили эти шаги на этапе установки!
- Прочитайте обновлённое руководство пользователя на GitHub!

**Comment by hakhagmon** [comment]
ftp сервера теперь нет из коробки?

**Reply by OSPanel** [reply]
Нет. Он морально устарел и уже почти нигде не используется, сейчас балом правит sftp. Скоро в сборку добавим ssh/sftp сервер, но это конечно не ftp. Подскажите пожалуйста, а для чего вам необходим именно FTP-сервер? Для каких операций он используется в вашем проекте?

**Reply by hakhagmon** [reply]
osp использую на отдельном пк. ssh/sftp меня вполне устроит :)

**Comment by Efess86** [comment]
Благодарю за ваш труд. 
Скажите пожалуйста а можно ли как-то вернуть старый интерфейс? Мне всего-то и нужно что ворпресс погонять а тут скачал 6-ю версию и как-то печально стало.
Спасибо за ответ.

**Reply by OSPanel** [reply]
Нет, мы делаем новый веб-интерфейс и не планируем возвращать старый.

**Comment by Sandmansss** [comment]
Подскажите пожалуйста как включить XDebug?
Я использую php 8.1.
В ".\config\PHP-8.1" я создал копию папки "default" и переименовал ее, внутри в "templates" в php.ini включил нужные мне параметры.
`xdebug.mode                    = "debug"
;xdebug.cli_color               = 0
;xdebug.client_discovery_header = "HTTP_X_FORWARDED_FOR,REMOTE_ADDR"
debug.client_host             = "localhost"
debug.client_port             = 9003`
`xdebug.idekey                   = "PHPSTORM"`
При запуске в модуле php 8.1 выбрал свой профиль. После этого перезапускаю openserver и ничего. 
В phpinfo я не вижу никаких изменений
Как включить эту чертову отладку????

**Reply by OSPanel** [reply]
Для включения расширения XDebug для PHP 8.1 в Open Server Panel, выполните следующие шаги:

1. **Откройте конфигурационный файл PHP 8.1:**

    Найдите папку, где установлен Open Server Panel, и перейдите в каталог с шаблонами настроек PHP 8.1. Обычно путь будет выглядеть следующим образом:

    ```
    <путь_к_установке_ospanel>\config\PHP-8.1\default\templates\php.ini
    ```

2. **Редактирование `php.ini`:**

    Откройте файл `php.ini` в текстовом редакторе (например, Notepad++ или VS Code).

3. **Раскомментируйте строку для загрузки XDebug:**

    Найдите строку, содержащую `zend_extension`, и убедитесь, что она не закомментирована (без точки с запятой в начале строки):

    ```ini
    zend_extension = xdebug
    ```

4. **Настройте параметры XDebug (опционально):**

    Это пример базовой конфигурации:

    ```ini
    [xdebug]
    xdebug.mode = debug
    xdebug.start_with_request = yes
    xdebug.client_host = "localhost"
    xdebug.client_port = 9003
    ```

    Вы можете настроить параметры в соответствии с вашими потребностями.

5. **Перезапуск Open Server Panel:**

    После внесения изменений в `php.ini` перезапустите модуль PHP, чтобы применить новые настройки. Для этого:
    
    - Щёлкните правой кнопкой мыши на значок Open Server Panel в области уведомлений.
    - Выберите необходимый модуль и нажмите **Перезапустить**.

6. **Проверка работы XDebug:**

    Чтобы убедиться, что XDebug включён и работает, создайте файл `info.php` в корневом каталоге вашего веб-проекта с содержимым:

    ```php
    <?php phpinfo(); ?>
    ```

    Откройте этот файл в браузере (например, `http://example.local/info.php`) и найдите секцию XDebug в выведенной информации. Если XDebug включён и настроен правильно, вы увидите информацию о XDebug.

После этих шагов расширение XDebug должно быть успешно активировано для PHP 8.1 в Open Server Panel.

**Reply by morovinger** [reply]
@OSPanel эта инструкция актуальна? у меня на версии 6,2,6 дебаг не включается если я включаю его в \modules\PHP-8.2\ospanel_data\default\templates\php.ini и разумеется слетает при перезагрузке если включаю напрямую в modules\PHP-8.2\PHP\php.ini

**Reply by OSPanel** [reply]
> @OSPanel эта инструкция актуальна? у меня на версии 6,2,6 дебаг не включается если я включаю его в \modules\PHP-8.2\ospanel_data\default\templates\php.ini и разумеется слетает при перезагрузке если включаю напрямую в modules\PHP-8.2\PHP\php.ini

Оба этих пути неверны, в первом случае вы пытались изменить дефолтный файл, который используется при восстановлении настроек до заводских, во втором случае вы редактировали временный файл, который пересоздаётся при каждом перезапуске PHP. Пожалуйста, прочитай документацию к программе, прежде чем пытаться изменять её конфигурацию.

**Comment by subonedev** [comment]
where is phpmyadmin ???

**Reply by OSPanel** [reply]
PHPMyAdmin is no longer included in Open Server Panel. Now, each user can decide which tool to use for database management.

You can install any suitable tool, for example:

- **PHPMyAdmin:** [Download from the official website](https://www.phpmyadmin.net/)
- **Adminer:** [Download from the official website](https://www.adminer.org/)
- **HeidiSQL:** [Download from the official website](https://www.heidisql.com/)
- **DBeaver:** [Download from the official website](https://dbeaver.io/)

We recommend choosing the tool that best meets your requirements and preferences. Installation and setup instructions can be found on the official websites of these tools.

**Reply by oscarhandsome** [reply]
very strange decision, if you can also include the default version or an opportunity to choose one as you did with all other parts, such as php versions or mysql etc.
